packages/jaeger-ui/src/model/ddg/PathElem.tsx

```diff
@@ -52,4 +52,42 @@ export default class PathElem {
     }
     return this._visibilityIdx;
   }
+
+  get focalSideNeighbor(): PathElem | null {
+    if (!this.distance) return null;
+    return this.memberOf.members[this.memberIdx - this.distance / Math.abs(this.distance)];
+  }
+
+  get focalSideEdgesKey(): TDdgEdgeIdentifier {
+    return this.distance < 0 ? 'egressEdges' : 'ingressEdges';
+  }
+
+  get farSideEdgesKey(): TDdgEdgeIdentifier {
+    return this.distance > 0 ? 'egressEdges' : 'ingressEdges';
+  }
+
+  private toJSONHelper = () => ({
+    memberIdx: this.memberIdx,
+    operation: this.operation.name,
+    service: this.operation.service.name,
+    visibilityIdx: this._visibilityIdx,
+  });
+
+  toJSON() {
+    return {
+      ...this.toJSONHelper(),
+      memberOf: {
+        focalIdx: this.memberOf.focalIdx,
+        members: this.memberOf.members.map(member => member.toJSONHelper()),
+      },
+    };
+  }
+
+  toString() {
+    return JSON.stringify(this.toJSON(), null, 2);
```

Comment:
Why override `toString` instead of returning this from `toJSON`?

Reply:
let's consolidate the toJSON / toString talk in [one discussion](https://github.com/jaegertracing/jaeger-ui/pull/386#discussion_r292510033)

Reply:
Doh! [toJSON](https://tc39.es/ecma262/#sec-serializejsonproperty)